### PR TITLE
Visitor can see different products on a restaurants menu 

### PR DESCRIPTION
**Pull request by kermit-klein**
[PT LINK](https://www.pivotaltracker.com/story/show/172642044)

Client can show menu list through axios get request(did test with mock request)
Client can connect with API to show plain text of menu items
cy:run script added for semaphore

### CODE REVIEW

File: cypress/integration/userCanSeeMenu.feature.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+describe("User can see menu", () => {
+  beforeEach(() => {
+    cy.server();
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/v1/menu_items",
+      response: "fixture:menu_list_index.json",
+    });
+  });
```
Comment (ViaMarcus) [inline cypress/integration/userCanSeeMenu.feature.js:9]: Should have newline

File: cypress/fixtures/menu_list_index.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "items": [
```
Comment (oliverochman) [inline cypress/fixtures/menu_list_index.json]: this should be the same as the model name in the backend

File: cypress/integration/userCanSeeMenu.feature.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  it("shows one menu item", () => {
+    cy.visit("/");
+    cy.get("#menuitem-1").within(() => {
+      cy.contains("Pizza");
+      cy.contains("Best pizza");
+      cy.contains("55");
+    });
+  });
+
+  it("shows one menu item", () => {
```
Comment (oliverochman) [inline cypress/integration/userCanSeeMenu.feature.js]: duplicate naming of it block

`it("shows another menu item"`

File: cypress/integration/userCanSeeMenu.feature.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    cy.visit("/");
+    cy.get("#menuitem-2").within(() => {
```
Comment (oliverochman) [inline cypress/integration/userCanSeeMenu.feature.js]: not a fan of "#menuitem"

please use "#menu-item" instead

File: src/components/ShowMenu.jsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React, { Component } from "react";
+import axios from "axios";
+
+export class ShowMenu extends Component {
+  state = {
+    menuList: [],
+  };
+
+  componentDidMount() {
+    let headers = {
```
Comment (oliverochman) [inline src/components/ShowMenu.jsx]: do not think you need to specify these headers if you do not add anything else to them

File: src/components/ShowMenu.jsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      "Content-type": "application/json",
+      Accept: "application/json",
+    };
+    axios.get("/menu_items", { headers: headers }).then((response) => {
```
Comment (oliverochman) [inline src/components/ShowMenu.jsx]: please use `try, catch` instead of using `.then`

old JS syntax

File: src/components/ShowMenu.jsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import React, { Component } from "react";
+import axios from "axios";
+
+export class ShowMenu extends Component {
```
Comment (oliverochman) [inline src/components/ShowMenu.jsx]: why `ShowMenu` and not just `Menu`?

File: src/components/ShowMenu.jsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        menuList: response.data.items,
+      });
+    });
+  }
+
+  render() {
+    const menuList = this.state.menuList;
+    let menu;
+    if (menuList.length > 0) {
+      menu = menuList.map((k) => {
```
Comment (oliverochman) [inline src/components/ShowMenu.jsx]: you need to change `k` to something more descriptive

perhaps `item`

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by oliverochman — APPROVED** [review]

**Pull request merged**